lib/evaluation.py

```diff
@@ -0,0 +1,106 @@
+from sklearn.metrics import *
+from abc import ABC, abstractmethod
+
+
+class Evaluator(ABC):
+	
+	def __init__(self, y_true, y_pred):
+		self.y_true = y_true
+		self.y_pred = y_pred
+
+	def data_is_valid(self) -> bool:
+		return len(self.y_true) == len(self.y_pred)
+
+	@abstractmethod
+	def evaluate(self) -> dict:
+		pass
+
+
+class TabularEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_pred):
+		super().__init__(y_true, y_pred)
+
+	def evaluate(self) -> dict:
+		pass
+
+
+class RegressionEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_pred):
+		super().__init__(y_true, y_pred)
+
+	def evaluate(self) -> dict:
+		
+		if not self.data_is_valid():
+			raise ValueError('Invalid Data')
+		
+		scores = {
+				'rootMeanSquaredError': mean_squared_error(self.y_true, self.y_pred, squared=False),
+				'meanAbsoluteError': mean_absolute_error(self.y_true, self.y_pred),
+				'meanAbsolutePercentageError': 'Infinity' if any(self.y_true == 0 | self.y_pred == 0) else
+				mean_absolute_percentage_error(self.y_true, self.y_pred),
+				'rSquared': r2_score(self.y_true, self.y_pred),
+				'rootMeanSquaredLogError': 'NaN' if any(self.y_true < 0 | self.y_pred < 0) else mean_squared_log_error(
+						self.y_true, self.y_pred),
+		}
+		
+		return scores
+
+
+class ClassificationEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_pred, thresholds, max_predictions):
+		super().__init__(y_true, y_pred)
+		self.thresholds = thresholds
+		self.max_predictions = max_predictions
+
+	def confidence_matrix(self, confidence_threshold) -> dict:
+		cm = confusion_matrix(self.y_true, self.y_pred)
+		tn = cm[0][0]
+		fn = cm[1][0]
+		tp = cm[1][1]
+		fp = cm[0][1]
+
+		confidence_matrix_for_threshold = {
+				'confidenceThreshold': confidence_threshold,
+				'maxPredictions': self.max_predictions,
+				'recall': recall_score(self.y_true, self.y_pred),
```

Comment:
Looks like that the confidenceThreshold will not work since it doesn't impact metrics like recall, precision, etc. as it should. Here is an example 
![image](https://user-images.githubusercontent.com/98159352/186426502-8ca8d743-329e-4692-a459-82e38ea1e1fc.png)

For detailed info, please check our OpenAPI doc about this field.

Reply:
btw, what kind of output should I expect? bagged probabilities with ppr? one-hot-encoded predictions?